### PR TITLE
#159053988 Add functionality for users to be able to comment on articles

### PR DESCRIPTION
**Pull request by hoslack**
**What does this PR do?**
- Allow users to make and delete comments

**Description of tasks completed**
- Create endpoints for CRUD operations of comments
- Create a  model for comments
- Make a column in the comments model for child comments(replies)
- Add unit tests for the comments functionality

**How can this PR be manually tested**
- Git pull the source code from ft-159053988-Comment-on-articles branch
- Copy the variables from `sample.env` and put the correct parameters in `.env` file
- Run `source .env` on your terminal to export environmental variables and activate virtual env
- Run `python manage.py migrate` to apply recent migrations
- Run `pip install -r requirements` to install latest dependencies
- Run `python manage.py runserver` to start the project
- If not already done, create a new user account and login
- Create an article and copy the slug
- Make a POST request to the following endpoint `api/v1/<str: slug>/comments/` replacing the `<slug>` with the slug you copied above. The data should be a comment 
`{"content": "Sample comment"}`
- You should get the comment back. 
- Make a DELETE request to the following endpoint `api/v1/<str: slug>/comments/<int: id>/` replacing the <id> with the id of the comment you got from the delete request you copied above. This returns 204_NO_CONTENT
- Make a GET request with the endpoint `api/v1/<str: slug>/comments/` and view all the comments you have made to the article from which you have taken the slug.

**Relevant Stories**
[159053988](https://www.pivotaltracker.com/n/projects/2184741/stories/159053988)

### CODE REVIEW

File: authors/apps/comments/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        Test that a user can't comment on a non-existent article.
+        """
+        comment_view = CommentsListCreateAPIView.as_view()
+        url = reverse('comments:all_comments', kwargs={"slug": '1234what'})
+        request = self.request_factory.post(url, data=self.sample_comment, format='json')
+        force_authenticate(request, user=self.user)
+        response = comment_view(request, slug='1234what')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_get_comment_from_article_without_comment(self):
```
Comment (RuthNjeri) [inline authors/apps/comments/tests/test_views.py:112]: The name of the function and what you have described in the docstrings that the function should do not add up.

File: authors/apps/comments/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        Test that a user can delete comment
+        """
+        pk = self.comment_response.data['id']
+        comment_view = CommentsCreateDeleteAPIView.as_view()
+        url = reverse('comments:comment_detail', kwargs={"slug": self.slug, "pk": pk})
+        request = self.request_factory.delete(url)
+        force_authenticate(request, user=self.user)
+        response = comment_view(request, slug=self.slug, pk=pk)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
```
Comment (RuthNjeri) [inline authors/apps/comments/tests/test_views.py:158]: Good job testing extensively.

**Comment by RuthNjeri** [comment]
The comments work well. Good job!

**Review by lennykioko — CHANGES_REQUESTED** [review]
I think it would be a good idea if we could allow users to update comments that they have made.

**Review by RuthNjeri — CHANGES_REQUESTED** [review]

**Review by lennykioko — APPROVED** [review]

**Comment by munyili-samuel** [comment]
I have tested and it looks ok to me. Good job

**Review by munyili-samuel — APPROVED** [review]
LGTM

**Review by RuthNjeri — APPROVED** [review]

**Pull request merged**